### DISCUSSION TITLE
auth_groups_users change group

### DISCUSSION THREAD

**Discussion by pippuccio76** [codeigniter4/shield#1058] [Q&A]
Hi there is a  way to change group without  :

```php
$user_group_list=$user->getGroups();

$user->removeGroup($user_group_list);

$user->addGroup($newgroup);
```

Symply updating ?

**Comment by datamweb** [comment]
What is output `$user_group_list`?

**Reply by pippuccio76** [reply]
An  array with group of user ( i set one group for one user):

![image](https://github.com/codeigniter4/shield/assets/5321982/60ae8da4-a981-4974-ac7b-fa167c262bfe)



tecnico is custom group

**Reply by datamweb** [reply]
And what is `$newgroup`?

**Reply by pippuccio76** [reply]
i create a db table to select user group ( one user one group) and select from form

**Reply by pippuccio76** [reply]
it works:

```
        //recupero l'id dell'utente nella tabella dei gruppi
        $gruppi_user=$user->getGroups();

        //dd($user_group_list);

        $user->removeGroup($gruppi_user[0]);

        $user->addGroup($post['group']);
```


But i want only update...

**Reply by datamweb** [reply]
What is the value of `$post['group']`?

**Reply by pippuccio76** [reply]
Value from table with my custom group

**Reply by datamweb** [reply]
What I understand is that you have a user from which you want to take group `tecnico` and add a new group to it, for this you can use the following:
```php
$user->syncGroups('your custom group value1', 'your custom group value 2');
```
See https://shield.codeigniter.com/references/authorization/#syncgroups
